routers/api/v1/repo/notes.go

```diff
@@ -58,8 +58,14 @@ func getNote(ctx *context.APIContext, identifier string) {
 		return
 	}
 
+	commitSHA, err := ctx.Repo.GitRepo.ConvertToSHA1(identifier)
+	if err != nil {
+		ctx.NotFound(err)
```

Comment:
`NotFound` on error?
Not `ServerError`?
I think we should differentiate between the two when possible.

Reply:
Yes, you're right. I added a check to differentiate error types.